src/Journal.tsx

```diff
@@ -46,19 +42,28 @@ export default class Journal extends React.PureComponent<{
 
     if (nextEntries) {
       const { hotDates } = this.state;
-      nextEntries.forEach((entry: Entry) => {
-        if (entry.entryTime) {
-          hotDates[entry.entryTime.toDate().toDateString()] =
-            entry.workout.color;
-        }
-      });
+      this.getColorsForDates(nextEntries, hotDates);
 
       this.setState(({ entries }: { entries: Array<Entry> }) => {
         return { entries: entries.concat(nextEntries), hotDates };
       });
     }
   };
 
+  getColorsForDates = (entries, hotDates) => {
```

Comment:
This function is mutating the `hotDates` object that is passed in. 

In general it is easier to debug (and test.....if we had testing) functions when they are pure (e.g. there's no side effects such as in-place mutations) because you can trust that the inputs aren't changed and the outputs are only the return values and nothing else. 

I would change this function to:
```
getColorsForDates = (entries) => {
    const hotDates = {};
    ...
    return hotDates
}
```

And then I would change the caller(s) of the function accordingly:
```
handleEntriesChange = (entries: Array<Entry>) => {
    const hotDates =  this.getColorsForDates(entries);
    this.setState({ entries, hotDates });
}
```

Reply:
This makes a lot of sense, function is now updated! Thanks for the detailed examples :D